src/components/tabs/types.ts

```diff
@@ -53,3 +52,42 @@ export interface TabsProps {
 // }
 
 // export type TabsProps = PropsForAriaLabel | PropsForAriaLabelledBy
+
+/**
+ * RawTabItem is used for items parsed directly from <Tab /> children
+ */
+export interface RawTabItem {
+  label: string
+  content: ReactNode
+}
+
+/**
+ * RawTabItemWithID includes tabId & panelId properties.
+ * These are regenerated at the same time as RawTabItem entries are parsed,
+ * so the separation here is for convenience and clarity.
+ */
+export interface RawTabItemWithIds extends RawTabItem {
+  tabId: string
+  panelId: string
+}
+
+/**
+ * TabItem includes all properties needed to render tabs,
+ * which includes active state.
+ */
+export interface TabItem extends RawTabItemWithIds {
+  isActive?: boolean
+}
+
+/**
+ * TabControlsProps is used for both TabButtonControls & TabDropdownControls.
+ * Those components are meant to be interchangeable, so they share
+ * the same interface.
+ */
+export interface TabControlsProps {
```

Comment:
I considered adding these types in type files within `components/tabs/components/...` directories, but the types are pretty well shared, so co-locating them all here felt more intuitive.